examples/stable_diffusion/train_dreambooth_loha.py

```diff
@@ -0,0 +1,1103 @@
+    )
+    parser.add_argument(
+        "--class_prompt",
+        type=str,
+        default=None,
+        help="The prompt to specify images in the same class as provided instance images.",
+    )
+    parser.add_argument(
+        "--with_prior_preservation",
+        default=False,
+        action="store_true",
+        help="Flag to add prior preservation loss.",
+    )
+    parser.add_argument("--prior_loss_weight", type=float, default=1.0, help="The weight of prior preservation loss.")
+    parser.add_argument(
+        "--num_class_images",
+        type=int,
+        default=100,
+        help=(
+            "Minimal class images for prior preservation loss. If there are not enough images already present in"
+            " class_data_dir, additional images will be sampled with class_prompt."
+        ),
+    )
+    parser.add_argument(
+        "--validation_prompt",
+        type=str,
+        default=None,
+        help="A prompt that is used during validation to verify that the model is learning.",
+    )
+    parser.add_argument(
+        "--num_validation_images",
+        type=int,
+        default=4,
+        help="Number of images that should be generated during validation with `validation_prompt`.",
+    )
+    parser.add_argument(
+        "--validation_steps",
+        type=int,
+        default=100,
+        help=(
+            "Run dreambooth validation every X steps. Dreambooth validation consists of running the prompt"
+            " `args.validation_prompt` multiple times: `args.num_validation_images`."
+        ),
+    )
+    parser.add_argument(
+        "--output_dir",
+        type=str,
+        default="text-inversion-model",
+        help="The output directory where the model predictions and checkpoints will be written.",
+    )
+    parser.add_argument("--seed", type=int, default=None, help="A seed for reproducible training.")
+    parser.add_argument(
+        "--resolution",
+        type=int,
+        default=512,
+        help=(
+            "The resolution for input images, all the images in the train/validation dataset will be resized to this"
+            " resolution"
+        ),
+    )
+    parser.add_argument(
+        "--center_crop", action="store_true", help="Whether to center crop images before resizing to resolution"
+    )
+    parser.add_argument("--train_text_encoder", action="store_true", help="Whether to train the text encoder")
+
+    # lora args
+    parser.add_argument("--use_loha", action="store_true", help="Whether to use LoHa for parameter efficient tuning")
+    parser.add_argument("--r", type=int, default=8, help="LoHa rank, only used if use_loha is True")
+    parser.add_argument("--alpha", type=int, default=32, help="LoHa alpha, only used if use_lora is True")
+    parser.add_argument("--rank_dropout", type=float, default=0.0, help="LoHa dropout for rank")
+    parser.add_argument("--module_dropout", type=float, default=0.0, help="LoHa dropout for disabling module at all")
```

Comment:
Where does `module_dropout` come from? At first glance, I couldn't find it described in the paper.

Reply:
As far as I get it, it's some heuristic, LyCORIS developers came up with (all their adapters have it). I left it to make code compatible with their implementation (but I am not sure, whether it is used by someone).

Probably we need more experiments to show that this parameter seems reasonable, but if you don't mind, I'll leave it for compatibility.

If we have some more data, maybe one day we could add it to all other adapters.